### PR TITLE
the start of user data getter

### PR DESCRIPTION
**Pull request by bthaile**
associated with https://github.com/AugurProject/augur/issues/3589

new getter to get user data
 * get user trading history and markets trading history, need both

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🍭

 This link will always point to this PR's latest changes: https://peek.run/5m3hejg

**Review by justinbarry — APPROVED** [review]

File: packages/augur-sdk/src/state/getter/Users.ts [hunk 1]
```diff
@@ -135,6 +142,12 @@ export interface ProfitLossResult {
   total: string;
 }
 
+export class UserAccountDataResult {
```
Comment (justinbarry) [inline packages/augur-sdk/src/state/getter/Users.ts]:
```suggestion
export interface UserAccountDataResult {
```

File: packages/augur-sdk/src/state/getter/Users.ts [hunk 2]
```diff
@@ -153,6 +166,35 @@ export class Users {
...
+    });
+    // collect marketIds
+    const marketIds = Object.keys(userTradeHistory);
+    const marketTradeHistory = await Trading.getTradingHistory(augur, db, { marketIds });
+    let marketsInfo = await Markets.getMarketsInfo(augur, db, { marketIds });
+    return {
+      userTradeHistory,
+      marketTradeHistory,
+      marketsInfo
+    } as UserAccountDataResult;
```
Comment (justinbarry) [inline packages/augur-sdk/src/state/getter/Users.ts]: Hopefully you won't need to coerce it with the above suggestion.

**Review by bthaile — COMMENTED** [review]
Made changes. Removed the explicit casting.

**Pull request merged**